### ISSUE TITLE
Installing using pip results in package resolution impossible error for pettingzoo and torch

### ISSUE THREAD

**Issue by almair404** [opendilab/DI-engine#614]
Hi, 
Working in an ubuntu 22.04 windows subsystem for linux (w11 box) using python 3.8.16. I tried used `pip install DI-engine`.

resulting in
```
ERROR: Cannot install di-engine==0.1.0, di-engine==0.1.1, di-engine==0.2.0, di-engine==0.2.1, di-engine==0.2.2, di-engine==0.2.3, di-engine==0.3.0, di-engine==0.3.1, di-engine==0.4.0, di-engine==0.4.1, di-engine==0.4.2, di-engine==0.4.3, di-engine==0.4.4, di-engine==0.4.5 and di-engine==0.4.6 because these package versions have conflicting dependencies.

The conflict is caused by:
    di-engine 0.4.6 depends on pettingzoo==1.12.0
    di-engine 0.4.5 depends on pettingzoo==1.12.0
    di-engine 0.4.4 depends on pettingzoo==1.12.0
    di-engine 0.4.3 depends on pettingzoo==1.12.0
    di-engine 0.4.2 depends on pettingzoo==1.12.0
    di-engine 0.4.1 depends on pettingzoo==1.12.0
    di-engine 0.4.0 depends on torch<=1.10.0 and >=1.1.0
    di-engine 0.3.1 depends on torch<=1.10.0 and >=1.1.0
    di-engine 0.3.0 depends on torch<=1.10.0 and >=1.1.0
    di-engine 0.2.3 depends on torch<=1.10.0 and >=1.1.0
    di-engine 0.2.2 depends on torch<=1.10.0 and >=1.1.0
    di-engine 0.2.1 depends on torch<=1.10.0 and >=1.1.0
    di-engine 0.2.0 depends on torch<=1.9.0 and >=1.3.1
    di-engine 0.1.1 depends on torch<=1.8.0 and >=1.3.1
    di-engine 0.1.0 depends on torch<=1.8.0 and >=1.3.1

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
```
If trying to install specific versions
```
$ pip install pettingzoo==1.12.0
ERROR: Could not find a version that satisfies the requirement pettingzoo==1.12.0 (from versions: 0.1.0, 0.1.1, 0.1.2, 0.1.3, 0.1.4, 0.1.5, 0.1.6, 0.1.7, 0.1.8, 0.1.9, 0.1.10, 0.1.11, 0.1.12, 0.1.13, 0.1.14, 1.0.0, 1.0.1, 1.1.0, 1.2.0, 1.2.1, 1.3.0, 1.3.1, 1.3.2, 1.3.3, 1.3.4, 1.3.5, 1.14.0, 1.15.0, 1.16.0, 1.17.0, 1.18.1, 1.19.0, 1.20.1, 1.21.0, 1.22.0, 1.22.1, 1.22.2, 1.22.3)
ERROR: No matching distribution found for pettingzoo==1.12.0
```
and
```
$ pip install torch==1.10.0
ERROR: Could not find a version that satisfies the requirement torch==1.10.0 (from versions: 1.11.0, 1.12.0, 1.12.1, 1.13.0, 1.13.1)
ERROR: No matching distribution found for torch==1.10.0
```

How to use pip to install DI-engine?

**Comment by PaParaZz1** [comment]
This bug is due to the compatible version problem of pettingzoo in ubuntu 22.04, we have fixed this problem in #597. You can install DI-engine from the latest main branch. And we will release this modification in the release of DI-engine v0.4.7.

**Comment by almair404** [comment]
Many thanks for the quick response and sorry i didn't dig around more for a solution. I installed the development version using the instructions given in the installation guide without problem.
